src/app/shared/components/product-card/product-card.component.html

```diff
@@ -0,0 +1,21 @@
+<article class="products__card card">
+  <div class="card__info">
+    <div class="card__description">
+      <div class="card__product-name">
+        <a [routerLink]="['/products/', productData.id]">{{ productData.name }}</a>
+      </div>
+      <div [productPriceColor]="productData.price"
+           class="card__product-price">{{ productData.price | uahCurrency}}</div>
+    </div>
+    <ng-content></ng-content>
+  </div>
+
+
+  <button (click)="toggleProduct()"
+          [ngClass]="{
+                      'btn__in-cart': inCart,
+                      '':!inCart
```

Comment:
Could you please explain this solution?

Reply:
inCart is a public variable of product-card.component.ts
The value depend on condition in 27 line. If some of products in cart has the same id as current product, inCard become true. This will change the class of button, that show the user state of the card according the cart